### ISSUE TITLE
Webcam causing Error

### ISSUE THREAD

**Issue by andrejpejovski** [C0D3D3V/bbb-dl#61]
[Here is the video am trying to download](https://bbb-lb.finki.ukim.mk/playback/presentation/2.3/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753?meetingId=dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753)

For some reason now it works even with webcam but i dont know the first time what happen. Can you help me to prevent this from happening again?

Here is the console log:

`
C:\Users\andre>bbb-dl --verbose --encoder hevc_nvenc --audiocodec mp3 -sw --crf -1 --preset 13 "https://bbb-lb.finki.ukim.mk/playback/presentation/2.3/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753?meetingId=dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753"
Output directory for the final video is: C:\Users\andre
Directory for the temporary files is: C:\Users\andre\AppData\Local\bbb-dl\90a13508ea4c196048af382b9948c393
Downloading meta information
metadata.xml is already present
shapes.svg is already present
Downloading webcams / deskshare
cursor.xml is already present
panzooms.xml is already present
captions.json is already present
deskshare.xml is already present
Downloading events.xml from: https://bbb-lb.finki.ukim.mk/presentation/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753/events.xml
presentation_text.json is already present
slides_new.xml is already present
Downloading notes.html from: https://bbb-lb.finki.ukim.mk/presentation/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753/notes.html
Downloading polls.json from: https://bbb-lb.finki.ukim.mk/presentation/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753/polls.json
Downloading external_videos.json from: https://bbb-lb.finki.ukim.mk/presentation/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753/external_videos.json
video/webcams.webm is already present
Downloading video/webcams.mp4 from: https://bbb-lb.finki.ukim.mk/presentation/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753/video/webcams.mp4
Downloading deskshare/deskshare.webm from: https://bbb-lb.finki.ukim.mk/presentation/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753/deskshare/deskshare.webm
Downloading deskshare/deskshare.mp4 from: https://bbb-lb.finki.ukim.mk/presentation/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753/deskshare/deskshare.mp4
notes.html could not be downloaded: 404 Not Found
Error: 404, message='Not Found', url=URL('https://bbb-lb.finki.ukim.mk/presentation/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753/notes.html')
events.xml could not be downloaded: 404 Not Found
Error: 404, message='Not Found', url=URL('https://bbb-lb.finki.ukim.mk/presentation/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753/events.xml')
external_videos.json could not be downloaded: 404 Not Found
Error: 404, message='Not Found', url=URL('https://bbb-lb.finki.ukim.mk/presentation/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753/external_videos.json')
video/webcams.mp4 could not be downloaded: 404 Not Found
Error: 404, message='Not Found', url=URL('https://bbb-lb.finki.ukim.mk/presentation/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753/video/webcams.mp4')
polls.json could not be downloaded: 404 Not Found
Error: 404, message='Not Found', url=URL('https://bbb-lb.finki.ukim.mk/presentation/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753/polls.json')
deskshare/deskshare.mp4 could not be downloaded: 404 Not Found
Error: 404, message='Not Found', url=URL('https://bbb-lb.finki.ukim.mk/presentation/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753/deskshare/deskshare.mp4')
deskshare/deskshare.webm could not be downloaded: 404 Not Found
Error: 404, message='Not Found', url=URL('https://bbb-lb.finki.ukim.mk/presentation/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753/deskshare/deskshare.webm')
Downloading slides
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-1.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-2.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-3.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-4.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-5.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-6.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-7.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-8.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-9.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-10.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-11.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-12.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-13.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-14.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-15.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-16.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-17.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-18.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-19.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-20.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-21.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-22.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-23.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-24.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-25.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-26.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-27.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-28.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-29.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-30.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-31.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-32.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-33.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-34.png is already present
presentation/77735affeb91e4f68759381cc1a9cee51e0466a9-1602661853914/slide-35.png is already present
Recording title: Глава 1
Recording date: 2020-10-14T09-50-04
Recording duration: 1:05:09
BBB version: 3.7.4+
No desk was shared in this session
Start capturing frames...
Output directory for frames is: C:\Users\andre\AppData\Local\bbb-dl\90a13508ea4c196048af382b9948c393\frames
Initialization takes a few seconds...

Done: 00000 / 00319 Frames | 000 / 039 Parts /
1/39 Partition already finished: 0 to 34

2/39 Partition already finished: 34 to 38

3/39 Partition already finished: 47 to 49

4/39 Partition already finished: 38 to 47

5/39 Partition already finished: 49 to 52

6/39 Partition already finished: 8:24 to 9:00

7/39 Partition already finished: 3:15 to 3:44

8/39 Partition already finished: 3:44 to 8:24

9/39 Partition already finished: 52 to 3:15

10/39 Partition already finished: 9:00 to 11:22
Done: 00068 / 00319 Frames | 010 / 039 Parts |
11/39 Partition already finished: 12:25 to 15:20

12/39 Partition already finished: 11:22 to 12:25

13/39 Partition already finished: 15:20 to 16:16

14/39 Partition already finished: 16:16 to 17:03

15/39 Partition already finished: 17:03 to 20:54

16/39 Partition already finished: 20:54 to 22:14

17/39 Partition already finished: 26:44 to 29:19

18/39 Partition already finished: 22:14 to 26:44

19/39 Partition already finished: 29:19 to 31:31

20/39 Partition already finished: 31:31 to 32:10

21/39 Partition already finished: 32:10 to 32:37

22/39 Partition already finished: 32:37 to 35:39

23/39 Partition already finished: 35:39 to 37:36

24/39 Partition already finished: 37:36 to 39:44

25/39 Partition already finished: 39:44 to 43:07

26/39 Partition already finished: 43:07 to 45:25

27/39 Partition already finished: 45:25 to 45:27

28/39 Partition already finished: 45:27 to 45:31

29/39 Partition already finished: 45:31 to 47:20

30/39 Partition already finished: 47:20 to 50:50
Done: 00276 / 00319 Frames | 030 / 039 Parts \
31/39 Partition already finished: 51:52 to 53:27

32/39 Partition already finished: 50:50 to 51:52

33/39 Partition already finished: 53:27 to 54:15

34/39 Partition already finished: 54:59 to 57:00

35/39 Partition already finished: 54:15 to 54:59

36/39 Partition already finished: 57:00 to 59:34

37/39 Partition already finished: 59:34 to 1:02:19

38/39 Partition already finished: 1:03:42 to 1:05:09

39/39 Partition already finished: 1:02:19 to 1:03:42

Frames capturing is finished and took: 3.
Start creating slideshow...
Slideshow does already exist! Skipping rendering!
Mux final slideshow
Running command: ffmpeg -hide_banner -i C:\Users\andre\AppData\Local\bbb-dl\90a13508ea4c196048af382b9948c393\video\webcams.webm -i C:\Users\andre\AppData\Local\bbb-dl\90a13508ea4c196048af382b9948c393\slideshow.mp4 -c:v hevc_nvenc -c:a mp3 -map 0:a -map 1:v -strict experimental -crf -1 -preset 13 -shortest C:\Users\andre\2020-10-14T09-50-04_Глава 1.mp4
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\andre\AppData\Roaming\Python\Python312\Scripts\bbb-dl.exe\__main__.py", line 7, in <module>
  File "C:\Users\andre\AppData\Roaming\Python\Python312\site-packages\bbb_dl\main.py", line 1479, in main
    bbb_dl.run()
  File "C:\Users\andre\AppData\Roaming\Python\Python312\site-packages\bbb_dl\main.py", line 276, in run
    result_path = self.final_mux(slideshow_path, webcams_path, webcams_rel_path, metadata)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\andre\AppData\Roaming\Python\Python312\site-packages\bbb_dl\main.py", line 1133, in final_mux
    asyncio.run(
  File "C:\Python312\Lib\asyncio\runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\asyncio\base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "C:\Users\andre\AppData\Roaming\Python\Python312\site-packages\bbb_dl\ffmpeg.py", line 323, in add_audio_to_slideshow
    await ffmpeg.execute()
  File "C:\Users\andre\AppData\Roaming\Python\Python312\site-packages\ffmpeg\asyncio\ffmpeg.py", line 208, in execute
    raise FFmpegError.create(message=tasks[2].result(), arguments=self.arguments)
ffmpeg.errors.FFmpegError: Error opening input files: Invalid data found when processing input`

**Comment by C0D3D3V** [comment]
Can you please run `ffmpeg -hide_banner -i C:\Users\andre\AppData\Local\bbb-dl\90a13508ea4c196048af382b9948c393\video\webcams.webm -i C:\Users\andre\AppData\Local\bbb-dl\90a13508ea4c196048af382b9948c393\slideshow.mp4 -c:v hevc_nvenc -c:a mp3 -map 0:a -map 1:v -strict experimental -crf -1 -preset 13 -shortest C:\Users\andre\2020-10-14T09-50-04_Глава 1.mp4`
In a terminal and send the output here?

**Comment by andrejpejovski** [comment]
This is very strange behaviour of bbb-dl. I use it very often and still can't figure it out what happend last night.

1. I runned the command as you said as you said and finished with success. [Here it is.](https://pastebin.com/QsyyggUp)

2. After this i deleted helper folder that bbb-dl stores slides, webcam, slideshow etc... also deleted the final outputvideo, and I runned the command again (same command from last night--> bbb-dl --verbose --encoder hevc_nvenc --audiocodec mp3 -sw --crf -1 --preset 13 "https://bbb-lb.finki.ukim.mk/playback/presentation/2.3/dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753?meetingId=dda8257dfa2d3b1a88d71054093e9e315e79da98-1602661804753") and it works just fine. But last night i was runninng it for 3-4 times and every time was crashing.

[Here is the console this time (second time) runned from scratch and it finished with success again](https://pastebin.com/jhEMUzSc)


I really don't know what was the problem last night. Using it bbb-dl very often and this was the stranges behavious of all times. Never happend before

**Comment by C0D3D3V** [comment]
🤷 Strange. But good that it now works.

Was this the first time, this happend? 

My best guess would be that, somhow the webcam.mp4 file was bugged, beacuse of an download error. It could be that I implemented a error in the method to continue downloading after Internet error, and so the final webcam file was corrupted.

But since you are the first one reporting such an error... 🤷 we have to wait for more data

**Comment by andrejpejovski** [comment]
Yes, first time happening.

Bro I found out. There's nothing wrong with the bbb-dl. It was my fault.

My Laptop mode was set to "optimized" instead of "standard" and I guess my graphics card was not running and in the bbb-dl i specifed that i want `--encoder hevc_nvenc`. I switched to optimized now and i got the same error as the first comment of this issue ticked. I switched to standard and the video finished successfully.

![image](https://github.com/user-attachments/assets/62fede72-dc38-4b47-a91a-c9f470b28a74)
![image](https://github.com/user-attachments/assets/924b14b3-488c-4ac4-808b-d03acb0edc26)
